api/globalv1/api.proto

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package globalv1;
+
+import "api/localv1/api.proto";
+
+option go_package = "sigs.k8s.io/kpng/api/globalv1";
+
+//
+// global (cluster-wide) model
+//
+
+message ServiceInfo {
+  uint64 Hash = 1;
+
+  localv1.Service Service = 2;
+}
+
+message EndpointInfo {
+  uint64 Hash = 1;
+
+  string Namespace = 2;
+  string SourceName = 3;
+  string ServiceName = 4;
+  reserved 5;
+  string PodName = 8;
+
+  localv1.Endpoint Endpoint = 6;
+  EndpointConditions Conditions = 7;
+  TopologyInfo Topology = 9;
+  TopologyHints Hints = 10;
+}
+
+message EndpointConditions {
```

Comment:
Can we get rid of the indirection if `Ready` is the only condition we care about?